resources/assets/components/CampaignInbox/index.js

```diff
@@ -65,6 +67,40 @@ class CampaignInbox extends React.Component {
     });
   }
 
+  updateQuantity(post, newQuantity) {
+    // Fields to send to /posts
+    const fields = {
+      northstar_id: post.user.id,
+      campaign_id: post.signup.campaign_id,
+      campaign_run_id: post.signup.campaign_run_id,
+      quantity: newQuantity,
+    };
+
+    // Make API request to Rogue to update the quantity on the backend
+    let response = this.api.post('api/v2/posts', fields);
+
+    response.then((result) => {
+      // Update the state
+      this.setState((previousState) => {
+        const newState = {...previousState};
+        const signupChanged = post.signup_id;
+
+        // Update the quantity for each post under this signup
+        forEach (newState.posts, (value) => {
+          if (value.signup_id == signupChanged) {
+            value.signup.quantity = newQuantity;
+          }
+        });
+
+        // Return the new state
+        return newState;
+      });
+    });
+
+    // Close the modal
+    this.hideHistory(event);
```

Comment:
oooo okay yeah. I removed `event` here and added a check in `hideHistory` to only use it if it's set